### PR TITLE
add of y_min, y_max and increasing options in isotonic_l2_pav

### PR DESCRIPTION
**Pull request by michaelsdr**

### CODE REVIEW

File: jaxopt/_src/isotonic.py [hunk 1]
```diff
@@ -100,15 +111,18 @@ def isotonic_l2_pav(y):
...
-        " To install, run 'pip install numba'."
-    )
-  # Define the expected shape & dtype of output.
-  shape_dtype = jax.ShapeDtypeStruct(shape=y.shape, dtype=y.dtype)
-  return jax.pure_callback(_isotonic_l2_pav, shape_dtype, y)
-
+  if increasing is True:
+    sign = -1
+  else:
+    sign = 1
```
Comment (mblondel) [inline jaxopt/_src/isotonic.py:117]: sign = -1 if increasing else 1

File: jaxopt/_src/isotonic.py [hunk 2]
```diff
@@ -100,15 +111,18 @@ def isotonic_l2_pav(y):
...
+  y = y * sign
+  sol = _isotonic_l2_pav(y) * sign
```
Comment (mblondel) [inline jaxopt/_src/isotonic.py:119]:
```suggestion
  sol = _isotonic_l2_pav(sign * y) * sign
```

File: jaxopt/_src/isotonic.py [hunk 3]
```diff
@@ -100,15 +111,18 @@ def isotonic_l2_pav(y):
...
+  if y_min is None:
+      y_min = -jnp.inf
+  if y_max is None:
+      y_max = jnp.inf
```
Comment (mblondel) [inline jaxopt/_src/isotonic.py:123]: You can set the default values directly in the function.

File: jaxopt/_src/isotonic.py [hunk 4]
```diff
@@ -89,7 +89,18 @@ def _isotonic_l2_pav(y):
...
+        "Numba could not be imported. Code will run much more slowly."
+        " To install, run 'pip install numba'."
+    )
+  # Define the expected shape & dtype of output.
+  shape_dtype = jax.ShapeDtypeStruct(shape=y.shape, dtype=y.dtype)
+  sol = jax.pure_callback(_isotonic_l2_pav_numba, shape_dtype, y)
+  return sol
+
+def isotonic_l2_pav(y, y_min=None, y_max=None, increasing=True):
   """Solves an isotonic regression problem using PAV.
```
Comment (mblondel) [inline jaxopt/_src/isotonic.py:105]: Document y_min, y_max and increasing (you need to write the equations in the two cases)

**Review by mblondel — COMMENTED** [review]
Thanks a lot Michael! Some comments below.

File: jaxopt/_src/isotonic.py (hunk 2 above)
Comment (michaelsdr) [reply jaxopt/_src/isotonic.py:119]: Done

File: jaxopt/_src/isotonic.py (hunk 3 above)
Comment (michaelsdr) [reply jaxopt/_src/isotonic.py:123]: Done

**Review by mblondel — APPROVED** [review]
LGTM

**Pull request merged**